File: tests/code/corpus/langs/scala/Shapes.scala
package geometry

import scala.math.Pi

trait Drawable {
  def draw(): Unit
}

class Shape extends Drawable {
  def draw(): Unit = {}
  def area(): Double = 0.0
}

class Circle(radius: Double) extends Shape {
  override def area(): Double = Pi * radius * radius
}

object Factory {
  def makeCircle(radius: Double): Circle = new Circle(radius)
}

def totalArea(shapes: List[Shape]): Double = shapes.map(s => s.area()).sum

File: tests/code/corpus/langs/kotlin/Shapes.kt
package geometry

import kotlin.math.PI

open class Shape {
    open fun area(): Double {
        return 0.0
    }

    fun describe(): String {
        return "shape of area " + area()
    }
}

class Circle(private val radius: Double) : Shape() {
    override fun area(): Double {
        return PI * radius * radius
    }
}

object Factory {
    fun makeCircle(radius: Double): Circle {
        return Circle(radius)
    }
}

fun totalArea(shapes: List<Shape>): Double {
    return shapes.sumOf { it.area() }
}
